pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentTopic.java

```diff
@@ -1231,37 +1232,53 @@ private void asyncDeleteCursorWithClearDelayedMessage(String subscriptionName,
             if (ex != null) {
                 unsubscribeFuture.completeExceptionally(ex);
             } else {
-                asyncDeleteCursor(subscriptionName, unsubscribeFuture);
+                asyncDeleteCursor(subscriptionName, persistentSubscription, unsubscribeFuture);
             }
         });
     }
 
-    private void asyncDeleteCursor(String subscriptionName, CompletableFuture<Void> unsubscribeFuture) {
-        ledger.asyncDeleteCursor(Codec.encode(subscriptionName), new DeleteCursorCallback() {
-            @Override
-            public void deleteCursorComplete(Object ctx) {
-                if (log.isDebugEnabled()) {
-                    log.debug("[{}][{}] Cursor deleted successfully", topic, subscriptionName);
-                }
-                removeSubscription(subscriptionName);
-                unsubscribeFuture.complete(null);
-                lastActive = System.nanoTime();
-            }
+    private void asyncDeleteCursor(String subscriptionName,
```

Comment:
we can remove `subscriptionName`, only use `subscription`

Reply:
Done.